src/config/parser.rs

```diff
@@ -118,6 +122,15 @@ impl Config {
             _ => parsed_cet,
         };
 
+        let proxy_str = globals.get::<_, String>("proxy")?;
+        let proxy = match Proxy::all(proxy_str) {
+            Ok(proxy) => Some(proxy),
+            Err(_) => {
+                log::error!("Invalid proxy url, defaulting to no proxy.");
+                None
+            }
+        };
```

Comment:
```suggestion
        #[cfg(feature = "socks")]
        let proxy_str = globals.get::<_, String>("proxy")?;
        #[cfg(feature = "socks")]
        let proxy = match Proxy::all(proxy_str) {
            Ok(proxy) => Some(proxy),
            Err(_) => {
                log::error!("Invalid proxy url, defaulting to no proxy.");
                None
            }
        };
```

I would suggest having this behind a `cfg` feature macro as this will only add the following code to the binary if the `socks` feature has been enabled during build otherwise it will not include it thus reducing binary size and removing unneeded code. 🙂